lib/compress/zstd_compress.c

```diff
@@ -1390,9 +1389,31 @@ ZSTD_reset_matchState(ZSTD_matchState_t* ms,
     }
 
     /* table Space */
-    DEBUGLOG(4, "reset table : %u", crp!=ZSTDcrp_noMemset);
+    DEBUGLOG(4, "reset table : %u", crp!=ZSTDcrp_noMemset && (crp == ZSTDcrp_forceMemset || cParams->strategy != ZSTD_fast));
     assert(((size_t)ptr & 3) == 0);  /* ensure ptr is properly aligned */
+#ifdef NORESET
+    if (crp!=ZSTDcrp_noMemset) {
+       size_t tableSpace;
+       if (cParams->strategy == ZSTD_fast && crp!=ZSTDcrp_forceMemset)  {
+           /* Randomize the offset table.
+              This is for testing purpose, this makes sure that ZSTD_fast compression
+              is robust to uninitialized table. */
```

Comment:
👍  excellent

Reply:
I really like this. I think we should leave it in and randomize the table whenever `ZSTD_DEBUG >= 1`.